src/libPMacc/include/cuSTL/container/allocator/DeviceMemAllocator.tpp

```diff
@@ -38,7 +38,7 @@ DeviceMemAllocator<Type, T_dim>::allocate(const math::Size_t<T_dim>& size)
 
     cudaData.ptr = NULL;
     cudaData.pitch = 1;
-    cudaData.xsize = size.x();
+    cudaData.xsize = size.x() * sizeof (Type);
```

Comment:
please check all usages of `cudaData.xsize` that they not multiplied by `sizeof (Type)`